graphstorm-processing/graphstorm_processing/repartition_files.py

```diff
@@ -861,197 +839,280 @@ def main():
-                            "Assigning re-partitioned feature files for '%s' to reverse "
-                            "edge type '%s'",
-                            feature_name,
-                            reverse_edge_type_name,
-                        )
-                        edge_data_meta[reverse_edge_type_name][feature_name] = feature_dict
-                    else:
-                        logging.debug(
-                            "Did not find reverse of edge type '%s', %s in %s",
-                            type_name,
-                            reverse_edge_type_name,
-                            edge_data_meta.keys(),
-                        )
-                else:
-                    logging.info(
-                        "Skipping repartitioning feature files for edge type, '%s',"
-                        " feature %d/%d '%s'",
-                        type_name,
-                        feature_idx + 1,
-                        len(type_data_dict),
-                        feature_name,
-                    )
-
-            # Indicate label and mask 1D arrays should be read in as flat arrays by dispatch_data.py
             if type_name in edge_types_with_labels:
                 # If the task is not link_prediction, we need to modify the label file's metadata
                 if task_type not in {"link_predict", "link_prediction"}:
-                    assert etype_label_property, (
+                    assert etype_label_property in type_data_dict, (
                         "When task is not link prediction, providing an 'etype_label_property' "
-                        f"is required. Got task {task_type}, but `metadata_dict['graph_info']"
-                        "['etype_label_property']` was empty."
+                        f"is required. Got task {task_type}, but {etype_label_property=}"
+                        f"was not in {type_data_dict=}"
                     )
+                    label_files = type_data_dict[etype_label_property]["data"]  # type: List[str]
                     logging.info(
-                        "Modifying metadata for label '%s' of edge type '%s'",
+                        "Modifying Parquet metadata for %d files of label '%s' of edge type '%s'",
+                        len(label_files),
                         etype_label_property,
                         type_name,
                     )
-                    label_files = type_data_dict[etype_label_property]["data"]  # type: List[str]
-                    # TODO: These ops are probably safe to parallelize, since the tables are likely
-                    # easy to fit in memory (one float/bool per row)
                     repartitioner.modify_metadata_for_flat_arrays(label_files)
                 for mask in ["train_mask", "test_mask", "val_mask"]:
-                    logging.info("Modifying metadata for '%s' of edge type '%s'", mask, type_name)
                     if mask in type_data_dict:
```

Comment:
What is the difference between re-partitioning edge features and re-partitioning edge masks?

Reply:
Is this because, masks are usually 1D?

Reply:
Can you also add a TODO here: when supporting multi-task training, it is possible that the masks may have other names like train_mask_a, train_mask_b, etc.

Reply:
Same for labels.

Reply:
> Is this because, masks are usually 1D?

Yes, DistPart expects label and mask arrays to be marked as flat using a metadata entry, so this function, `modify_flat_array_metadata` does that.

Will add the TODO.